pkg/utils/erofs/erofs.go

```diff
@@ -12,12 +12,22 @@ import (
 	"golang.org/x/sys/unix"
 )
 
-func Mount(bootstrapPath, fsID, mountPoint string) error {
+func Mount(bootstrapPath, fsID, shareDomainID, mountPoint string) error {
 	mount := unix.Mount
+	var opts string
 
-	opts := "fsid=" + fsID
+	if shareDomainID != "" {
+		opts = fmt.Sprintf("domain_id=%s,fsid=%s", shareDomainID, fsID)
+	} else {
+		opts = "fsid=" + fsID
+	}
 	logrus.Infof("Mount erofs to %s with options %s", mountPoint, opts)
+
 	if err := mount("erofs", mountPoint, "erofs", 0, opts); err != nil {
+		if errors.Is(err, unix.EINVAL) && shareDomainID != "" {
```

Comment:
Well. Snapshotter won't retry if syscall `mount` returns EINVAL?

Reply:
Yup, if user's system doesn't support shared domain, an error will be returned directly an won't do fallback mount anymore.